src/include/common/file_system/file_system.h

```diff
@@ -43,6 +43,12 @@ class KUZU_API FileSystem {
 
     virtual bool canHandleFile(const std::string& /*path*/) const { KU_UNREACHABLE; }
 
+    std::string pathSeparator();
```

Comment:
I would suggest not adding more functions to the FileSystem class since any changes to the fileSystem class may require re-compilation of extensions.